### ISSUE TITLE
option to use mesos-containerizer

### ISSUE THREAD

**Issue by sedninja** [MustWin/mesos-memcached#1]
This looks like it could be useful for anyone in need of a scalable memcached cluster, but I'm curious what your thoughts are regarding the use of the default mesos-containerizer for the memcached instances vs docker. It seems like you could get more for your money if you didn't have to use the docker container with it's extra resource requirements over a simple jailed memcached instance. I love docker for applications with a lot of dependencies, but installing memcached on the hosts and then running the host binary seems more appealing to me personally.

I am just getting into go so I don't know how much I can contribute but I am going to go ahead and fork this project and see what I can do.

**Comment by mikejihbe** [comment]
I actually agree with you, it would be nice to ditch the docker dependency.
To do so we need to figure out how to get the right memcached binary
downloaded, and I'd like to maintain the dynamic port assignment.
If you want to get that going that'd be great!

On Tuesday, December 29, 2015, sedninja notifications@github.com wrote:

> This looks like it could be useful for anyone in need of a scalable
> memcached cluster, but I'm curious what your thoughts are regarding the use
> of the default mesos-containerizer for the memcached instances vs docker.
> It seems like you could get more for your money if you didn't have to use
> the docker container with it's extra resource requirements over a simple
> jailed memcached instance. I love docker for applications with a lot of
> dependencies, but installing memcached on the hosts and then running the
> host binary seems more appealing to me personally.
> 
> I am just getting into go so I don't know how much I can contribute but I
> am going to go ahead and fork this project and see what I can do.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/MustWin/mesos-memcached/issues/1.

## 

Sent from an abacus

**Comment by mikejihbe** [comment]
And figure out the best way to inspect the memcached instance, probably the
memcached stats interface, but maybe something mesos has built in would
work too.

On Tuesday, December 29, 2015, Mike Ihbe mikejihbe@gmail.com wrote:

> I actually agree with you, it would be nice to ditch the docker
> dependency. To do so we need to figure out how to get the right
> memcached binary downloaded, and I'd like to maintain the dynamic port
> assignment.
> If you want to get that going that'd be great!
> 
> On Tuesday, December 29, 2015, sedninja <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote:
> 
> > This looks like it could be useful for anyone in need of a scalable
> > memcached cluster, but I'm curious what your thoughts are regarding the use
> > of the default mesos-containerizer for the memcached instances vs docker.
> > It seems like you could get more for your money if you didn't have to use
> > the docker container with it's extra resource requirements over a simple
> > jailed memcached instance. I love docker for applications with a lot of
> > dependencies, but installing memcached on the hosts and then running the
> > host binary seems more appealing to me personally.
> > 
> > I am just getting into go so I don't know how much I can contribute but I
> > am going to go ahead and fork this project and see what I can do.
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/MustWin/mesos-memcached/issues/1.
> 
> ## 
> 
> Sent from an abacus

## 

Sent from an abacus

**Comment by sedninja** [comment]
I agree that port assignment should be dynamic. I don't think anything needs to change there. I am working on a bit of code now to try and work out which interfaces should be used in the `-l` flag for memcached because we don't want to bother with localhost or the bridged adapters from docker if docker is running on the host. I will commit my changes once I have something working.

If there isn't anything in mesos to get stats for us, we can probably pull some of the logic from the memcache-top utility written in perl [here](https://github.com/eculver/memcache-top.git).
